pkg/operator/apis/monitoring/v1/types.go

```diff
@@ -1405,6 +1417,201 @@ type NodeMonitoring struct {
 	Spec NodeMonitoringSpec `json:"spec"`
 }
 
+// NodeScrapeEndpoint specifies a node's Prometheus metrics endpoint to scrape.
+type NodeScrapeEndpoint struct {
```

Comment:
* Hm.. I am missing "port" 🤔 
* Somehow interval is required?
![image](https://github.com/GoogleCloudPlatform/prometheus-engine/assets/6950331/2f7b12c9-f0e7-49fb-9760-99e56667274f)
* Scheme (you can totally use TLS) and basic HTTPClientConfig without secrets would be nice for a start 🤔 (to at least skip verify?)
* Params we could bring back as well, there is little harm

Reply:
- I removed port, since it's not part of the [prometheus metadata labels for nodes. ](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#node) Do you see users wanting to user port?
- If IIUC the Kubelet serves metrics over HTTPs, so I removed it to reduce the chance of a user error and simplify the code. I guess we can return error if it's not https. The kubelet has it's ca. So we need the HTTPClientConfig to look like this:
![image](https://github.com/GoogleCloudPlatform/prometheus-engine/assets/89922129/cc37b022-92b7-46aa-a120-a31824c5d64b)
Should we expose HTTPClientConfig and just block those fields?
- Yeah, I can bring it back.

Reply:
Got it, sounds like you want to tailor it to kubelet. However given "NodeMonitoring" semantics that might be surprising 🤔 I think I would be ok to start with tailored kubelet approach you propose, wonder what's @pintohutch opinion.

Reply:
Discussed with @macxamin offline - IIUC we want to keep port number and http as options. Port name however isn't applicable for the node as it's not in the [spec](https://pkg.go.dev/k8s.io/api/core/v1#NodeDaemonEndpoints).

Reply:
Yes, I will add port logic in a future PR. I want to get nodeMonitoring to a working state and then I can make refinements to it.